src/toolkits/drawing_classifier/dc_data_iterator.cpp

```diff
@@ -0,0 +1,177 @@
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include <model_server/extensions/additional_sframe_utilities.hpp>
+#include <toolkits/drawing_classifier/dc_data_iterator.hpp>
+
+#include <algorithm>
+#include <cmath>
+
+#include <core/data/image/io.hpp>
+#include <core/data/flexible_type/flexible_type.hpp>
+#include <core/logging/logger.hpp>
+
+namespace turi {
+namespace drawing_classifier {
+
+using neural_net::shared_float_array;
+
+simple_data_iterator::target_properties
+simple_data_iterator::compute_properties(
+    const gl_sarray& targets,
+    std::vector<std::string> expected_class_labels) {
+
+  target_properties result;
+
+  // Determine the list of unique class labels,
+  gl_sarray classes = targets.unique().sort();
+
+  if (expected_class_labels.empty()) {
+
+    // Infer the class-to-index map from the observed labels.
+    result.classes.reserve(classes.size());
+    int i = 0;
+    for (const flexible_type& label : classes.range_iterator()) {
+      result.classes.push_back(label);
+      result.class_to_index_map[label] = i++;
+    }
+  } else {
+
+    // Construct the class-to-index map from the expected labels.
+    result.classes = std::move(expected_class_labels);
+    int i = 0;
+    for (const std::string& label : result.classes) {
+      result.class_to_index_map[label] = i++;
+    }
+
+    // Use the map to verify that we only encountered expected labels.
+    for (const flexible_type& ft : classes.range_iterator()) {
+      std::string label(ft);  // Ensures correct overload resolution below.
+      if (result.class_to_index_map.count(label) == 0) {
+        log_and_throw("Targets contained unexpected class label " + label);
+      }
+    }
+  }
+  return result;
+}
+
+simple_data_iterator::simple_data_iterator(const parameters& params)
+  : data_(params.data),
+
+    // Determine which column is which within each (ordered) row.
+    target_index_(data_.column_index(params.target_column_name)),
+    predictions_index_(params.predictions_column_name.empty()
+                       ? -1
+                       : data_.column_index(params.predictions_column_name)),
+    feature_index_(data_.column_index(params.feature_column_name)),
+
+    // Whether to traverse the SFrame more than once, and whether to shuffle.
+    repeat_(params.repeat),
+    shuffle_(params.shuffle),
+
+    // Identify/verify the class labels and other target properties.
+    target_properties_(
+        compute_properties(data_[params.target_column_name],
+                           params.class_labels)),
+
+    // Start an iteration through the entire SFrame.
+    range_iterator_(data_.range_iterator()),
+    next_row_(range_iterator_.begin()),
+
+    // Initialize random number generator.
+    random_engine_(params.random_seed)
+
+{}
+
+void add_drawing_pixel_data_to_batch(
+    float *next_drawing_pointer, const flex_image &bitmap) {
+  image_load_to_numpy(bitmap, (size_t)next_drawing_pointer, 
```

Comment:
Please use C++11 `static_cast` here

https://google.github.io/styleguide/cppguide.html#Casting

Reply:
Also, consider using the `copy_image_to_memory` function that @shreyajain17  is adding. https://github.com/apple/turicreate/pull/2421

Less confusing, since numpy is irrelevant/misleading here

Reply:
Added a call to `copy_image_to_memory`, which makes this PR dependent on #2421 .